exoscale/resource_exoscale_instance_pool.go

```diff
@@ -0,0 +1,487 @@
+
+	resp, err := client.RequestWithContext(ctx, egoscale.ListZones{})
+	if err != nil {
+		return false, err
+	}
+	zones := resp.(*egoscale.ListZonesResponse).Zone
+
+	for _, zone := range zones {
+		_, err := getInstancePoolByName(ctx, client, d.Id(), zone.ID)
+		if err != nil {
+			continue
+		}
+
+		return true, nil
+	}
+
+	return false, fmt.Errorf("Instance pool %q not found", d.Id())
+}
+
+func resourceInstancePoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning update", resourceInstancePoolIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutUpdate))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	id, err := egoscale.ParseUUID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	zone, err := getZoneByName(ctx, client, d.Get("zone").(string))
+	if err != nil {
+		return err
+	}
+
+	req := &egoscale.UpdateInstancePool{
+		ID:     id,
+		ZoneID: zone.ID,
+	}
+
+	if d.HasChange("description") {
+		req.Description = d.Get("description").(string)
+	}
+
+	var userData string
+	if d.HasChange("user_data") {
+		userData = base64.StdEncoding.EncodeToString([]byte(d.Get("user_data").(string)))
+		req.UserData = userData
+	}
+
+	_, err = client.RequestWithContext(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("size") {
+		scale := &egoscale.ScaleInstancePool{
+			ID:     id,
+			ZoneID: zone.ID,
+			Size:   d.Get("size").(int),
+		}
+
+		_, err = client.RequestWithContext(ctx, scale)
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] %s: update finished successfully", resourceInstancePoolIDString(d))
+
+	return resourceInstancePoolRead(d, meta)
+}
+
+func resourceInstancePoolDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning delete", resourceInstancePoolIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutDelete))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	id, err := egoscale.ParseUUID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	zone, err := getZoneByName(ctx, client, d.Get("zone").(string))
+	if err != nil {
+		return err
+	}
+
+	req := &egoscale.DestroyInstancePool{
+		ID:     id,
+		ZoneID: zone.ID,
+	}
+
+	if _, err := client.RequestWithContext(ctx, req); err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: delete finished successfully", resourceInstancePoolIDString(d))
+
+	return nil
+}
+
+func resourceInstancePoolImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutRead))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	resp, err := client.RequestWithContext(ctx, egoscale.ListZones{})
+	if err != nil {
+		return nil, err
+	}
+	zones := resp.(*egoscale.ListZonesResponse).Zone
+
+	resources := make([]*schema.ResourceData, 0, 1)
+	resources = append(resources, d)
```

Comment:
Since you don't append any more items to this slice later in this function, it's far more simpler and easier to read with `resources := []*schema.ResourceData{d}` IMHO. No need to meddle with the obscure `make()` parameters here 😏